tests/unit/lib/Any_unittest.cc

```diff
@@ -691,3 +691,13 @@ TEST(Any, lazy_cloneable_visitable)
 	EXPECT_FALSE(any.visitable());
 	EXPECT_THROW(any.visit(nullptr), std::logic_error);
 }
+
+TEST(Any, compare_string_vectors)
+{
+	std::string str("some string");
+	std::vector<std::string> lhs{str};
+	std::vector<std::string> rhs{str};
```

Comment:
sure, we can register and we can do equals, and what happens is that the == operator of std::vector is called. This one uses the == operator on all elements, https://en.cppreference.com/w/cpp/container/vector/operator_cmp

It will not work recursively, i.e. std::vector<SGVector<X>> will also use the == operator of the elements, which in shogun is pointer equality and not .equals